pkg/kubelet/status/manager.go

```diff
@@ -434,13 +435,19 @@ func (m *manager) needsReconcile(uid types.UID, status api.PodStatus) bool {
 		pod = mirrorPod
 	}
 
-	if isStatusEqual(&pod.Status, &status) {
+	podStatus, err := copyStatus(&pod.Status)
+	if err != nil {
+		return false
+	}
+	normalizeStatus(&podStatus)
```

Comment:
The normalization also sorts the container statuses though, which could be different from the API server.